nixos/modules/system/boot/systemd/tmpfiles.nix

```diff
@@ -163,6 +163,19 @@ in
   };
 
   config = {
+    warnings =
+      let
+        paths = lib.foldlAttrs (warnings: path: _:
+          warnings ++ lib.optional (lib.hasPrefix "/etc/tmpfiles.d/" path) path
+        ) [] config.boot.initrd.systemd.contents;
```

Comment:
We don't usually worry too much about eval performance, but this is very inefficient :P It should probably be more like `lib.filter (...) (builtins.attrNames config.boot.initrd.systemd.contents)`.

Also, it's technically more correct to filter based on `boot.initrd.systemd.storePaths.*.target`. Since the sytsemd 256 update, `contents` is just a convenience wrapper around `storePaths` that automatically sets `target`.

Reply:
Right, I had a more complicated expression first while figuring out the structure of the options and then I missed that the fold was not needed anymore, thanks for catching this!

I updated it now and also changed it to use `storePaths`.